### DISCUSSION TITLE
ping results not working as I would expect, but maybe I should be using differently

### DISCUSSION THREAD

**Discussion by half-metal** [kartikk221/hyper-express#56] [General]
Hi I am attempting to use ping method with websockets from the docs, however, it continually gets undefined returned:
```
ping(String|Buffer|ArrayBuffer: message): Sends a ping control message.
Returns Boolean[true] if message was sent successfully.
Returns Boolean[false] if message could not be sent due to built up backpressure.
```
When I use ws.send with 'ping' it works as expected. But I was expecting a boolean true or false if ws.ping was used. Below is example for how I was using comparing send vs ping
```
  async function wsHeartbeat(){
      console.log('Running wsHeartbeat')
        Object.keys(connections).forEach((id) => {
        connections[id].send('ping');
        pingResults = connections[id].ping('altping');
        console.log('pingResults are:',pingResults)
    });
  }
```
This is with two client connections where I see the ws.send ping on the clients, so would expect client websocket protocol to return pong results from the ws.ping method. Appreciate any guidance
![image](https://user-images.githubusercontent.com/58837299/155024865-a41810cb-60d8-4783-bf2a-eb02cf898a95.png)

**Comment by kartikk221** [comment]
Hi, this was a bug in the ping/various other Websocket methods which were performing the correct operation but not returning the proper boolean type as specified in the documentation. Some methods actually did not support the boolean property yet under the hood but I have just went back and confirmed everything with the uWebsockets.js documentation and implemented ternary operators to ensure all methods return the expected types.

Update to v5.8.2 to see if your problem is resolved.

**Comment by half-metal** [comment]
Nice, updated to latest version and got the Boolean response as shown in this image for ping
![image](https://user-images.githubusercontent.com/58837299/155213516-7f6ea65a-ce46-4d45-9485-c5a70874c99a.png)
